solutions/usaco-786.mdx

```diff
@@ -240,4 +240,48 @@ public class lifeguards {
 ```
 
 </JavaSection>
+
+<PySection>
+
+```python
+sweep = []
+	
+with open('lifeguards.in') as r:
+	n = int(r.readline())
+	for _ in range(n):
+		a, b = map(int, r.readline().split())
+		sweep.append([a, b])
+
+if n == 1:
+	print("0")
+	exit()
+
+sweep.sort()
+	
+cur = sweep[0][1]
+cover = sweep[0][1] - sweep[0][0]
+
+find_min = [0] * n
+
+for i in range(len(sweep)):
+	find_min[i] = sweep[i][1] - sweep[i][0]
+
+cnt = 1
+
+for start, end in sweep[1:]:
+	if cur > start:
+		cover += max(0, end - cur)
+		find_min[cnt-1] -= (cur - start) 
+		find_min[cnt] -= min(cur, end) - start
+	else: 
+		cover += end - start
+
+	cur = max(end, cur)
+	cnt += 1
+
+ans = cover - max(min(find_min), 0)
+
+print(str(ans))
```

Comment:
The `print` function automatically converts the argument a string, so `print(ans)` is fine.